src/appleseed/foundation/image/analysis.cpp

```diff
@@ -157,28 +145,23 @@ double compute_rms_deviation(const Image& image1, const Image& image2)
 
     double mse = 0.0;   // mean square error
```

Comment:
I think making `mse` a double is a very sensible choice. That doesn't mean that `compute_error_squared()` should return a double too.